src/mono/wasm/runtime/run.ts

```diff
@@ -15,13 +20,33 @@ export async function mono_run_main_and_exit(main_assembly_name: string, args: s
     }
 }
 
+/**
+ * Possible signatures are described here  https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/main-command-line
+ */
 export async function mono_run_main(main_assembly_name: string, args: string[]): Promise<number> {
     mono_wasm_set_main_args(main_assembly_name, args);
     if (runtimeHelpers.wait_for_debugger == -1) {
         console.log("MONO_WASM: waiting for debugger...");
-        return await mono_wasm_wait_for_debugger().then(() => mono_call_assembly_entry_point(main_assembly_name, [args], "m"));
+        await mono_wasm_wait_for_debugger();
     }
-    return mono_call_assembly_entry_point(main_assembly_name, [args], "m");
+    const method = find_entry_point(main_assembly_name);
+    return runtimeHelpers.javaScriptExports.call_entry_point(method, args);
+}
+
+export function find_entry_point(assembly: string) {
+    mono_assert(runtimeHelpers.mono_wasm_bindings_is_ready, "Expected binding to be initialized later during startup sequence.");
```

Comment:
this error message is confusing, is it asserting that the bindings were already initialized? if so the message should probably be different, i can suggest a different one if necessary